File: tests/journal.lib/ansi_emulates.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// type aliases
using ansi_t = pyre::journal::ansi_t;


// verify that we can detect ANSI emulation reliably; the harness sets things up and passes a
// flag on the command line so we can set our expectations
int main(int argc, char *argv[]) {

    // the test harness hands its expectation
    bool expectation = argc > 1 ? std::atoi(argv[1]) : true;
    // ask the wrapper whether the underlying terminal type is ANSI compatible
    auto observation = ansi_t::compatible();

    // verify that expectation and observation match
    assert (expectation == observation);

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/ansi_emulates.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


def test():
    """
    Verify that we can detect ANSI compatible terminals correctly
    """
    # for the command line
    import sys
    # access the color map
    from journal.ANSI import ANSI

    # pull the expected value from the command line
    expectation = bool(int(sys.argv[1])) if len(sys.argv) > 1 else True
    # ask the color map
    observation = ANSI.compatible()

    # verify that the two match
    assert expectation == observation

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
